### ISSUE TITLE
endpoint classification?

### ISSUE THREAD

**Issue by flokli** [francoposa/tower-otel-http-metrics#14]
Looking at the docs, we currently use a single `.layer()` call on a axum Router.

That way we however are not able to distinguish individual types of requests.

I'd ideally like to have the counters/histograms provided by this crate on a per-endpoint-of-specific-type granularity.

What do you think about adding an example applying the handler to two separate individual handlers?

We maybe would end up with separate `HTTPMetricsLayer` instances, and add a method to `HTTPMetricsLayerBuilder` allowing to add additional (static for the lifetime of `HTTPMetricsLayer`) fields?

**Comment by flokli** [comment]
The field to distinguish is probably `http.route`, as per https://opentelemetry.io/docs/specs/semconv/attributes-registry/http/

**Comment by francoposa** [comment]
Does this not cover the same goal? Have not tested it out yet but I figured there was likely a way to do this in a web framework without having to clone the layer implementations. https://docs.rs/axum/latest/axum/struct.Router.html#method.route_layer

**Comment by flokli** [comment]
Yes, but I'm not sure we can access the specific route string, didn't dig deep enough.

**Comment by francoposa** [comment]
I am not sure I understand the meaning or the goal of

> I'd ideally like to have the counters/histograms provided by this crate on a per-endpoint-of-specific-type granularity.

**Comment by flokli** [comment]
I have different types of routes, doing different things, taking different amount of time.

I'm proposing a mode in which these metrics are bucketed by the route in the web framework used (see the example next to `http.route`). So I can monitor response times of a specific endpoint.

**Comment by francoposa** [comment]
Still not clear on what exactly you want to do differently for the different endpoints - do you want to set different `ExplicitBucketBoundaries` for different endpoints?

Otherwise the endpoints are already labeled & distinguished by the `http.route`.

**Comment by flokli** [comment]
Sorry, I was just being stupid. Enabling the `axum` feature was all I needed to get the `http.route` fields to be added. Sorry for the noise, and thanks!
